frontend_tests/casper_tests/05-subscriptions.js

```diff
@@ -131,25 +131,43 @@ casper.then(function () {
 });
 casper.then(function () {
     casper.click('#create_stream_button');
-    casper.fill('form#stream_creation_form', {stream_name: '  '});
-    casper.click('form#stream_creation_form button.btn.btn-primary');
+    casper.waitUntilVisible("form#stream_creation_form", function () {
+        casper.fill('form#stream_creation_form', {stream_name: '  '});
+        casper.click('form#stream_creation_form button.btn.btn-primary');
+    });
 });
 casper.then(function () {
     casper.waitForSelectorText('#stream_name_error', 'A stream needs to have a name', function () {
         casper.test.assertTextExists('A stream needs to have a name', "Can't create a stream with an empty name");
-        casper.click('form#stream_creation_form button.btn.btn-default');
-        casper.fill('form#add_new_subscription', {stream_name: '  '});
-        casper.click('#create_stream_button');
-        casper.fill('form#stream_creation_form', {stream_name: 'Waseemio'});
-        casper.click('form#stream_creation_form button.btn.btn-primary');
+        casper.waitUntilVisible("form#stream_creation_form", function () {
+            casper.click('form#stream_creation_form button.btn.btn-default');
+            casper.fill('form#add_new_subscription', {stream_name: '  '});
+            casper.click('#create_stream_button');
+        });
+        casper.waitUntilVisible("form#stream_creation_form", function () {
+            casper.fill('form#stream_creation_form', {stream_name: 'Waseemio@'});
+            casper.click('form#stream_creation_form button.btn.btn-primary');
+        });
     });
 });
 casper.then(function () {
+   casper.waitForSelectorText('#stream_name_error', 'A stream needs to have valid characters', function () {
+       casper.test.assertTextExists('A stream needs to have valid characters', "Can't create a stream with invalid characters");
+        casper.click('#create_stream_button');
+        casper.waitUntilVisible("form#stream_creation_form", function () {
+            casper.fill('form#stream_creation_form', {stream_name: 'Waseemio'});
+            casper.click('form#stream_creation_form button.btn.btn-primary');
+        });
+   });
+});
+casper.then(function () {
     casper.waitForSelectorText('#stream_name_error', 'A stream with this name already exists', function () {
         casper.test.assertTextExists('A stream with this name already exists', "Can't create a stream with a duplicate name");
         casper.test.info('Streams should be filtered when typing in the create box');
-        casper.click('form#stream_creation_form button.btn.btn-default');
-    });
+        casper.waitUntilVisible("form#stream_creation_form", function () {
```

Comment:
When we get an error, the form closes? If it doesn't then we won't need this wait.

Reply:
No, the form doesn't close. I'll change this!